### DISCUSSION TITLE
[plugin] Cannot use Tool Tree in theia

### DISCUSSION THREAD

**Discussion by meng-jpg** [eclipse-theia/theia#14638] [Q&A] [answered]
I installed the extension in theia, but cannot use it. It works fine in vscode and the tool `ripgrep` has already installed. What should I do to fix the problem?
<img width="1025" alt="截屏2024-12-17 17 10 55" src="https://github.com/user-attachments/assets/69c764c9-fa06-41eb-825d-164069f6d6d6" />
theia version: electron 1.57.0
Operation System: macOS Sequoia 15.1.1

**Comment by msujew** [comment, marked as answer]
Hey @meng-jpg,

You can set the `todo-tree.ripgrep` setting to `<theia-install-directory>/resources/app/lib/backend/native/ripgrep`. We use ripgrep ourselves, but in a different directory than in vscode, so the extension cannot find it.

**Reply by meng-jpg** [reply]
> Hey @meng-jpg,
> 
> You can set the `todo-tree.ripgrep` setting to `<theia-install-directory>/resources/app/lib/backend/native/ripgrep`. We use ripgrep ourselves, but in a different directory than in vscode, so the extension cannot find it.

It really worked. Thank you very much for your reply.

**Reply by sdirix** [reply]
@msujew Would it make sense that we create a link to ripgrep wherever VSCode is placing it?

**Reply by msujew** [reply]
Sure, that might make sense. See [here](https://github.com/Gruntfuggly/todo-tree/blob/a6f60e0ce830c4649ac34fc05e5a1799ec91d151/src/config.js#L100-L110) for the logic that the extension is using to figure out any deployed ripgrep instance.

**Reply by meng-jpg** [reply]
That's good news, when can I expect to see this change in the releases?
